cmd/reset/reset-consumer-group-offset_test.go

```diff
@@ -7,6 +7,129 @@ import (
+
+	group := testutil.CreateConsumerGroup(t, "reset-cgo-multi", topicA, topicB)
+
+	testutil.ProduceMessage(t, topicA, "test-key", "test-value1", 0, 0)
+	testutil.ProduceMessage(t, topicA, "test-key", "test-value2", 0, 1)
+
+	testutil.ProduceMessage(t, topicB, "test-key", "test-value1", 0, 0)
+	testutil.ProduceMessage(t, topicB, "test-key", "test-value2", 0, 1)
+
+	kafkaCtl := testutil.CreateKafkaCtlCommand()
+	kafkaCtl.Verbose = false
+
+	if _, err := kafkaCtl.Execute("consume", topicA, "--group", group, "--max-messages", "2"); err != nil {
+		t.Fatalf("failed to execute command: %v", err)
+	}
+
+	consumed := strings.Split(strings.TrimSpace(kafkaCtl.GetStdOut()), "\n")
+	testutil.AssertContains(t, "test-value1", consumed)
+	testutil.AssertContains(t, "test-value2", consumed)
+
+	testutil.VerifyConsumerGroupOffset(t, group, topicA, 2)
+	testutil.VerifyConsumerGroupOffset(t, group, topicB, 0)
+
+	if _, err := kafkaCtl.Execute("reset", "consumer-group-offset", group, "--topic", topicA, "--topic", topicB, "--oldest", "--execute"); err != nil {
+		t.Fatalf("failed to execute command: %v", err)
+	}
+
+	testutil.VerifyConsumerGroupOffset(t, group, topicA, 0)
+	testutil.VerifyConsumerGroupOffset(t, group, topicB, 0)
+
+	kafkaCtl = testutil.CreateKafkaCtlCommand()
+	if _, err := kafkaCtl.Execute("reset", "consumer-group-offset", group, "--topic", topicA, "--topic", topicB, "--newest", "--execute"); err != nil {
+		t.Fatalf("failed to execute command: %v", err)
+	}
+
+	testutil.VerifyConsumerGroupOffset(t, group, topicA, 2)
+	testutil.VerifyConsumerGroupOffset(t, group, topicA, 2)
+}
+
+func TestResetCGOForAllTopicsInTheGroupIntegration(t *testing.T) {
+
+	testutil.StartIntegrationTest(t)
+
+	topicA := testutil.CreateTopic(t, "reset-cgo-a")
+	topicB := testutil.CreateTopic(t, "reset-cgo-b")
+	topicOther := testutil.CreateTopic(t, "reset-cgo-other")
+
+	group := testutil.CreateConsumerGroup(t, "reset-cgo-all", topicA, topicB)
+
+	testutil.ProduceMessage(t, topicA, "test-key", "test-value1", 0, 0)
+	testutil.ProduceMessage(t, topicA, "test-key", "test-value2", 0, 1)
+
+	testutil.ProduceMessage(t, topicB, "test-key", "test-value1", 0, 0)
+	testutil.ProduceMessage(t, topicB, "test-key", "test-value2", 0, 1)
+
+	testutil.ProduceMessage(t, topicOther, "test-key", "test-value1", 0, 0)
+	testutil.ProduceMessage(t, topicOther, "test-key", "test-value2", 0, 1)
+
+	testutil.VerifyConsumerGroupOffset(t, group, topicA, 0)
+	testutil.VerifyConsumerGroupOffset(t, group, topicB, 0)
+	testutil.VerifyTopicNotInConsumerGroup(t, group, topicOther)
+
+	kafkaCtl := testutil.CreateKafkaCtlCommand()
+
+	if _, err := kafkaCtl.Execute("reset", "consumer-group-offset", group, "--all-topics", "--newest", "--execute"); err != nil {
+		t.Fatalf("failed to execute command: %v", err)
+	}
+
+	testutil.VerifyConsumerGroupOffset(t, group, topicA, 2)
+	testutil.VerifyConsumerGroupOffset(t, group, topicB, 2)
+	testutil.VerifyTopicNotInConsumerGroup(t, group, topicOther)
+}
+
```

Comment:
since the consumer-group-offset reset did not have proper integration tests I added that on your PR directly.

Reply:
Thanks